jsapp/js/components/modalForms/formMedia.es6

```diff
@@ -0,0 +1,292 @@
+import {actions} from 'js/actions';
+import {bem} from 'js/bem';
+import {LoadingSpinner} from 'js/ui';
+
+import {
+  truncateString,
+  truncateUrl,
+} from 'js/utils';
+
+const MAX_ITEM_LENGTH = 50;
+const DEFAULT_MEDIA_DESCRIPTION = 'default';
+
+/**
+ * @prop {object} asset
+ *
+ * Modal for uploading form media
+ */
+class FormMedia extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      uploadedAssets: [],
+      fieldsErrors: {},
+      inputURL: '',
+      // to show loading icon instead of nothing on first load
+      isInitialised: false,
+      // to show loading icon while uploading any file
+      isUploadFilePending: false,
+      isUploadURLPending: false
+    };
+
+    autoBind(this);
+  }
+
+  /*
+   * setup
+   */
+
+  componentDidMount() {
+    actions.media.loadMedia(this.props.asset.uid);
+
+    actions.media.loadMedia.completed.listen(this.onGetMediaCompleted);
+    actions.media.uploadMedia.failed.listen(this.onUploadFailed);
+  }
+
+  /*
+   * action listeners
+   */
+
+  onGetMediaCompleted(uploadedAssets) {
+    this.setState({
+      uploadedAssets: uploadedAssets.results,
+      isUploadFilePending: false,
+      isUploadURLPending: false,
+      isInitialised: true,
+    });
+  }
+
+  onUploadFailed(response) {
+    this.setState({
+      fieldsErrors: response.responseJSON,
+      isUploadFilePending: false,
+      isUploadURLPending: false
+    });
+  }
+
+  /*
+   * Utilities
+   */
+
+  toBase64(file) {
+    return new Promise((resolve, reject) => {
+      var reader = new FileReader();
+      reader.onload = () => resolve(reader.result);
+      reader.onerror = reject;
+      reader.readAsDataURL(file);
+    });
+  }
+
+  /*
+   * Backend requires form media payload to include the following
+   *
+   * @namespace formMediaJSON
+   * @param {string} description - can be anything, when in doubt use 'default'
+   * @param {string} filetype - should be `ASSET_FILE_TYPES.form_media.id`
+   * @param {string} metadata - JSON stringified filename
+   * @param {string} base64Encoded
+   *
+   * @param {formMediaJSON} formMediaJSON
+   *
+   * Taken from: https://github.com/kobotoolbox/kpi/blob/a5302750fa0974d075495bfc070afc85ff5cf60d/kpi/views/v2/asset_file.py#L49-L67
+   */
+
+  uploadMedia(formMediaJSON) {
+    // Reset error message before uploading again
+    this.setState({fieldsErrors: {}});
+    actions.media.uploadMedia(this.props.asset.uid, formMediaJSON);
+  }
+
+  /*
+   * DOM listeners
+   */
+
+  onFileDrop(files) {
+    if (files.length >= 1) {
+      this.setState({isUploadFilePending: true});
+
+      files.forEach(async (file) => {
+        var base64File = await this.toBase64(file);
+
+        this.uploadMedia({
+	        description: DEFAULT_MEDIA_DESCRIPTION,
+          file_type: ASSET_FILE_TYPES.form_media.id,
+          metadata: JSON.stringify({filename: file.name}),
+	        base64Encoded: base64File
+	      });
```

Comment:
These lines have still wrong indentation :)

Reply:
Doesn't show up on my editor :confused: just redid the spacing